examples/snippets/dataset/manage/migrate_dataset_changes.py

```diff
@@ -0,0 +1,50 @@
+"""
+An example script to migrate changes to a dataset attributes from one instance of Tamr to another
+"""
+import tamr_toolbox as tbox
+
+# load example multi config
+my_config = tbox.utils.config.from_yaml("examples/resources/conf/migrate_dataset.config.yaml")
+
+# Use the configuration to create a global logger
+LOGGER = tbox.utils.logger.create(__name__, log_directory=my_config["logging_dir"])
+
+# Create the source tamr client
+source_client = tbox.utils.client.create(**my_config["source_migration_instance"])
+
+# Create the target tamr client
+target_client = tbox.utils.client.create(**my_config["target_migration_instance"])
+
+# Update each dataset
+datasets = my_config["datasets"]
+for ds in datasets:
+    dataset_name = ds["dataset_name"]
+
+    # Get dataset from source instance
+    source_dataset = source_client.datasets.by_name(dataset_name)
+
+    # Get updated dataset definition
+    attributes = [attr for attr in source_dataset.attributes.stream()]
+    attr_type_dict = {}
+    for attr in attributes:
+        attr_type_dict[attr.name] = attr.spec().to_dict()["type"]
+    attribute_names = attr_type_dict.keys()
```

Comment:
This can be more concise and Pythonic with a dictionary comprehension.  If you don't need `attributes` itself that can be skipped (see my other comment below).  It's also not necessary to use `stream()` when iterating over an `AttributeCollection`.
```suggestion
    # Get updated dataset definition
    attr_type_dict = {attr.name: attr.spec().to_dict()["type"] for attr in source_dataset.attributes}
    attribute_names = attr_type_dict.keys()
```

Reply:
I like this change although I'm interested in best practices for when to make one liners vs separating lines of code out to be more readable/maintainable

Reply:
Readability is a very good thing to consider.  Here I think the "Pythonic" solution just takes a little getting used to, but isn't really less clear than the loop expression.  I would most strongly advocate for list/dictionary comprehensions like this, with the addition of a single clear comment like `# Construct dictionary mapping attribute names to their types`.

When writing code where a nested comprehension is possible, but becomes hard to read, you can always separate it into multiple lines , using a generator expression `(my_function(x) for x in my_list)`.